utils/jwt.js

```diff
@@ -17,6 +17,7 @@ const isTokenValid = ({ token }) => {
 const attachCookiesToResponse = ({ res, user }) => {
   const token = createJWT({ payload: user });
 
+  // AKOSREVIEW: It would probably make sense to expire the cookie and the token at the same time
   const oneDay = 1000 * 60 * 60 * 24; //1d because we set previously JWT_LIFETIME=1d
```

Comment:
I think it would make sense to have the cookie and the json web token expire in the same lifetime; unless the instructor gave a reason to do it this way. 

`oneDay = process.env.JWT_LIFETIME`

though it may depend on how exactly you're defining `JWT_LIFETIME` ; is it a string, a number?

Reply:
@akosasante in .env it's a string
![image](https://github.com/NatachaKey/ecommerce/assets/103755090/fb88ab63-35cc-4869-abe5-4aa26c1c11a7)
 I thought they both are equal to 1d , just in this case the instructor showed how he calculated it in  miliseconds.
also I've tried  the following 
![image](https://github.com/NatachaKey/ecommerce/assets/103755090/517131a0-4fa7-405a-897a-26a420ea4037)

but if i set in.env JWT_LIFETIME in miliseconds it will also be a string and i will need to convert it to a number like here ![image](https://github.com/NatachaKey/ecommerce/assets/103755090/fde18b7e-2c7d-4faa-83bd-53780825a571)

I wonder if there is a way to set a number directly in .env?

Reply:
Ah yes I see the dillemma. But your approach works very well :+1: 

All values in environment variables (like in .env file) are always passed to node as strings. So you would always need to parse them in the code if you want a different type. 

Another way would be to use a JSON string in the env file:

`JWT_LIFETIME='{ "ms": 86400000, "string": "1d" }'`

Then in the code:

```javascript
const authLifetime = JSON.parse(process.env.JWT_LIFETIME);
const cookieExpiryOneDay = authLifetime.ms;
const jwtExpiryString = authLifetime.string;
```